src/general/MathUtils.cs

```diff
@@ -131,4 +131,31 @@ public static int PositiveModulo(this int val, int mod)
         double standardDeviation = Math.Sqrt(sumOfSquares / count - average * average);
         return (average, standardDeviation);
     }
+
+    /// <summary>
+    ///   Rounds the given value with decimals given sparingly.
+    /// </summary>
+    /// <param name="value">The value to be rounded.</param>
+    /// <param name="threshold">
+    ///   If <paramref name="value"/> is lower than this integer threshold, decimals will start being used.
+    /// </param>
+    public static double RoundWithMinimalDecimals(double value, int threshold)
+    {
+        double result;
+
+        if (value > threshold)
+        {
+            result = Math.Round(value);
+        }
+        else if (value > 0.1f)
+        {
+            result = Math.Round(value, 1);
+        }
+        else
+        {
+            result = Math.Round(value, 3);
+        }
+
+        return (float)result;
```

Comment:
There's an extra `(float)` cast here.

Reply:
Also this method could just return the result instead of having a separate variable for it in each of the if cases.